### PR TITLE
Refactor(web-react): Introduce `ModalCloseButton` component

### PR DESCRIPTION
**Pull request by literat**
## Description

This contribution has been, somehow, lost.

### Issue reference

https://github.com/lmc-eu/spirit-design-system/pull/1046

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-react/src/components/Modal/ModalCloseButton.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { ModalCloseButtonProps } from '../../types';
+import { Icon } from '../Icon';
+import { VisuallyHidden } from '../VisuallyHidden';
+import { Button } from '../Button';
+
+const ModalCloseButton = ({ label = 'Close', onClose, id, isOpen, ...restProps }: ModalCloseButtonProps) => (
```
Comment (crishpeen) [inline packages/web-react/src/components/Modal/ModalCloseButton.tsx:7]: no test for this subcomponent?

Comment (literat) [reply packages/web-react/src/components/Modal/ModalCloseButton.tsx:7]: You are damn right. Added.

File: packages/web-react/src/components/Modal/ModalHeader.tsx [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
 import React from 'react';
 import { useStyleProps } from '../../hooks';
-import { ClickEvent, ModalHeaderProps } from '../../types';
-import { Button } from '../Button';
-import { Icon } from '../Icon';
-import { VisuallyHidden } from '../VisuallyHidden';
+import { ModalHeaderProps } from '../../types';
 import { useModalContext } from './ModalContext';
 import { useModalStyleProps } from './useModalStyleProps';
+import ModalCloseButton from './ModalCloseButton';
```
Comment (crishpeen) [inline packages/web-react/src/components/Modal/ModalHeader.tsx:7]: 🔠

File: packages/web-react/src/types/modal.ts [hunk 3]
```diff
@@ -17,6 +17,11 @@ export type ModalDialogHandlingProps = {
   onClose: (event: ClickEvent) => void;
 };
 
+export interface ModalCloseButtonProps extends ModalDialogHandlingProps {
+  label: string;
+  id: string;
```
Comment (crishpeen) [inline packages/web-react/src/types/modal.ts:22]: abc?

**Review by crishpeen — APPROVED** [review]

**Review by pavelklibani — COMMENTED** [review]
Shouldn't be this component also exported in the index?

**Review by pavelklibani — APPROVED** [review]

File: packages/web-react/src/components/Modal/ModalHeader.tsx (hunk 2 above)
Comment (literat) [reply packages/web-react/src/components/Modal/ModalHeader.tsx]: Fixed.

File: packages/web-react/src/types/modal.ts (hunk 3 above)
Comment (literat) [reply packages/web-react/src/types/modal.ts:21]: Fixed.

**Comment by literat** [comment]
> Shouldn't be this component also exported in the index?

I have added the export. Thanks.

**Pull request merged**